docs/CppInteropability/GettingStartedWithC++Interop.md

```diff
@@ -0,0 +1,209 @@
+#  Getting started with C++ Interoperability
+
+This document is desgined to get you started with bidirectional API-level interoperability between Swift and C++.
+
+## Table of Contents
+
+- [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code)
+- [Adding C++ to an Xcode project](#adding-c-to-an-xcode-project)
+- [Creating a Swift Package](#Creating-a-Swift-Package)
+- [Building with CMake](#building-with-cmake)
+
+## Creating a Module to contain your C++ source code
+
+- Create a new C++ implementation and header file
+- For this example we will call the files Cxx, so we should have a Cxx.cpp and Cxx.hpp.
+- Next create an empty file and call it `module.modulemap`, in this file create the module for your source code, and define your C++ header (`requires cplusplus` isn't required but it's convention for C++ modules, especially if they use C++ features).
+
+```
+//In module.modulemap
+module Cxx {
+    //note that your header should be the file that containts your method implementations
+    header "Cxx.hpp"
+    requires cplusplus
+}
+```
+
+- Move the newly created files (Cxx.cpp, Cxx.hpp, module.modulemap) into a separate directory (this should remain in your project directory)
+
+<img width="252" alt="Screen Shot 2022-02-26 at 9 14 06 PM" src="https://user-images.githubusercontent.com/62521716/155867937-9d9d6c62-4418-414d-bc4e-5d12c2055022.png">
+
+## Adding C++ to an Xcode project
+- In your xcode project, follow the steps [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code) in your project directory
+
+Add the C++ module to the include path and enable C++ interop:
+- Navigate to your project directory 
+- In `Project` navigate to `Build Settings` -> `Swift Compiler`
+- Under `Custom Flags` -> `Other Swift Flags` add`-Xfrontend -enable-cxx-interop`
+- Under `Search Paths` -> `Import Paths` add your search path to the C++ module (i.e, `./ProjectName/Cxx`).
```

Comment:
Question: here i found that adding `-Xfrontend -enable-cxx-interop` to the `Targert` produced a lot of erros, however i still needed to add `./ProjectName/Cxx` to the `Import Paths`

is this expected?

If so i was thinking of updating to line 38:
- Under `Search Paths` -> `Import Paths` add your search path to the C++ module (i.e, `./ProjectName/Cxx`). Repeat this step in `Other Swift Flags`

Reply:
Sounds good to me. You can also do this as a follow up PR if you want, since the tests are almost finished. Whatever you'd rather.

Reply:
Cool cool, i will opt for an additional PR lol